packages/scaffold-ui/src/views/w3m-connecting-farcaster-view/index.ts

```diff
@@ -0,0 +1,227 @@
+      return html`${this.mobileTemplate()}`
+    }
+
+    return html`${this.desktopTemplate()}`
+  }
+
+  private desktopTemplate() {
+    return html` <wui-flex
+      flexDirection="column"
+      alignItems="center"
+      .padding=${['0', 'xl', 'xl', 'xl']}
+      gap="xl"
+    >
+      <wui-shimmer borderRadius="l" width="100%"> ${this.qrCodeTemplate()} </wui-shimmer>
+
+      <wui-text variant="paragraph-500" color="fg-100">
+        Scan this QR Code with your phone
+      </wui-text>
+      ${this.copyTemplate()}
+    </wui-flex>`
+  }
+
+  private mobileTemplate() {
+    return html` <wui-flex
+      flexDirection="column"
+      alignItems="center"
+      .padding=${['3xl', 'xl', 'xl', 'xl'] as const}
+      gap="xl"
+    >
+      <wui-flex justifyContent="center" alignItems="center">
+        <wui-logo logo="farcaster"></wui-logo>
+        ${this.loaderTemplate()}
+        <wui-icon-box
+          backgroundColor="error-100"
+          background="opaque"
+          iconColor="error-100"
+          icon="close"
+          size="sm"
+          border
+          borderColor="wui-color-bg-125"
+        ></wui-icon-box>
+      </wui-flex>
+      <wui-flex flexDirection="column" alignItems="center" gap="xs">
+        <wui-text align="center" variant="paragraph-500" color="fg-100"
+          >Continue in Farcaster</span></wui-text
+        >
+        <wui-text align="center" variant="small-400" color="fg-200"
+          >Accept connection request in the app</wui-text
+        ></wui-flex
+      >
+      ${this.mobileLinkTemplate()}
+    </wui-flex>`
+  }
+
+  private loaderTemplate() {
+    const borderRadiusMaster = ThemeController.state.themeVariables['--w3m-border-radius-master']
+    const radius = borderRadiusMaster ? parseInt(borderRadiusMaster.replace('px', ''), 10) : 4
+
+    return html`<wui-loading-thumbnail radius=${radius * 9}></wui-loading-thumbnail>`
+  }
+
+  private async connectFarcaster() {
+    if (this.authConnector) {
+      try {
+        await this.authConnector?.provider.connectFarcaster()
+
+        if (this.socialProvider) {
+          StorageUtil.setConnectedSocialProvider(this.socialProvider)
+        }
+        SnackController.showLoading('Loading user data')
+        this.loading = true
+        await ConnectionController.connectExternal(this.authConnector)
+        this.loading = false
+        ModalController.close()
+      } catch (error) {
+        RouterController.goBack()
+        SnackController.showError(error)
+      }
+    }
+  }
+
+  private mobileLinkTemplate() {
+    return html`<wui-button
+      size="md"
+      ?loading=${this.loading}
+      ?disabled=${!this.uri || this.loading}
+      @click=${() => {
+        if (this.uri) {
+          CoreHelperUtil.openHref(this.uri, '_blank')
+        }
+      }}
+    >
+      Open farcaster</wui-button
+    >`
+  }
+
+  private onRenderProxy() {
+    if (!this.ready && this.uri) {
+      this.timeout = setTimeout(() => {
```

Comment:
Cannot find where I did comment similar but same applies here: if we use `setTimeout` to resolve a problem, IMO we should add comments on top of it to describe why we use this solution. Bc later when we maintain the code, `setTimeout`s are hard to understand why they are used.